.github/workflows/release-file.yml

```diff
@@ -0,0 +1,50 @@
+name: Create Release File
+
+on:
+  release:
+    types: [published]
+  workflow_dispatch:
+
+jobs:
+  upload_archive:
+    name: Upload archive
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      id-token: write
+      attestations: write
+    steps:
+      - run: |
+          ref=${{ github.ref_name }}
```

Comment:
We would suggest enclosing the expansion here in double quotes, as a precaution. ie `ref="${{ github.ref_name }}"` - that way if a ref is created that would otherwise cause extra assignments or insert spaces here, it won't cause problems.